pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -179,37 +230,82 @@ func (tc *Controller) handleErr(err error, key interface{}) {
 		return
 	}
 
-	utilruntime.HandleError(fmt.Errorf("error cleaning up Job %v, will retry: %v", key, err))
+	utilruntime.HandleError(fmt.Errorf("error cleaning up Job/Pod %v, will retry: %v", key, err))
 	tc.queue.AddRateLimited(key)
 }
 
-// processJob will check the Job's state and TTL and delete the Job when it
-// finishes and its TTL after finished has expired. If the Job hasn't finished or
-// its TTL hasn't expired, it will be added to the queue after the TTL is expected
-// to expire.
+// processResource will check the Resource's [job / pod] state and TTL and
+// delete the resource when it finishes and its TTL after finished has expired.
+// If the resource hasn't finished or its TTL hasn't expired, it will be added
+// to the queue after the TTL is expected to expire.
 // This function is not meant to be invoked concurrently with the same key.
-func (tc *Controller) processJob(key string) error {
+func (tc *Controller) processResource(key string) error {
 	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		return err
 	}
 
-	klog.V(4).Infof("Checking if Job %s/%s is ready for cleanup", namespace, name)
-	// Ignore the Jobs that are already deleted or being deleted, or the ones that don't need clean up.
+	// check if the resource is Job
 	job, err := tc.jLister.Jobs(namespace).Get(name)
```

Comment:
It's possible that the resource is a Pod instead of a Job, and there happens to be a Job with the same name.